### ISSUE TITLE
export an API: linter.validate()

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#10718]
following #10716 

**The version of ESLint you are using.**
5.2.0/master

**The problem you want to solve.**
it can be useful to a tool author -- validate a config, without running linting.

**Your take on the correct solution to problem.**

```js
linter.validate(config, source);
```
atm the function throws an error, if something going wrong. is there a possibility to return an array of errors?
